pkg/lib/indexer/indexer.go

```diff
@@ -513,40 +523,68 @@ func (i ImageIndexer) ExportFromIndex(request ExportFromIndexRequest) error {
 	}
 
 	var errs []error
-	for _, bundleImage := range bundles {
-		// try to name the folder
-		folderName, err := dbQuerier.GetBundleVersion(context.TODO(), bundleImage)
-		if err != nil {
-			return err
-		}
-		if folderName == "" {
-			// operator-registry does not care about the folder name
-			folderName = bundleImage
-		}
-		exporter := bundle.NewExporterForBundle(bundleImage, filepath.Join(request.DownloadPath, folderName), request.ContainerTool)
-		if err := exporter.Export(); err != nil {
-			err = fmt.Errorf("error exporting bundle from image: %s", err)
-			errs = append(errs, err)
-		}
-	}
-	if err != nil {
-		errs = append(errs, err)
+	var wg sync.WaitGroup
+	wg.Add(len(bundles))
+	var mu = &sync.Mutex{}
+
+	sem := make(chan struct{}, concurrencyLimitForExport)
+
+	for bundleImage, bundleDir := range bundles {
+		go func(bundleImage string, bundleDir bundleDirPrefix) {
+			defer wg.Done()
+
+			sem <- struct{}{}
+			defer func() {
+				<-sem
```

Comment:
to make this airtight, I think we want a select that checks a context for cancellation, so don't wait forever for slow exports:

```go
select {
case <- ctx.Done():
case <- sem:
}
```

or we could just use [x/sync/semaphore](https://godoc.org/golang.org/x/sync/semaphore#Weighted.Acquire):

```go
import "golang.org/x/sync/semaphore"
// ...
sem := semaphore.NewWeighted(concurrencyLimitForExport)
//...
if err := sem.Acquire(ctx, 1); err != nil {
    return err
}
//...
```